### PR TITLE
Stardoc generates extras.md

### PR DESCRIPTION
**Pull request by Cvarier**
**What type of PR is this?**

> Documentation

**What does this PR do? Why is it needed?**
This is a mirror of the content in extras.rst, allowing us to manually inspect the delta between them:
LHS: https://github.com/bazelbuild/rules_go/blob/master/go/extras.rst
RHS: https://github.com/bazelbuild/rules_go/compare/master...aspect-dev:stardoc-chai#diff-facb0b3aafba052f3e5fe16fbe02d2c2f6b2c2c30c443fd08f4617d4c5d38532

As soon as we are satisfied that the generated content is correct, we'll delete extras.rst and update any links to it.
(This could happen as part of this PR, or subsequent)

Technique: 
1. Converted RST to Github-flavored MD using pandoc (https://pandoc.org/try)
2. Filled in gaps by copying doc strings out of existing RST file into the starlark files (converting to MD syntax).

First step in addressing https://github.com/bazelbuild/rules_go/issues/2437
Remaining steps will be to rinse-and-repeat for other .rst files containing rule APIs

**Which issues(s) does this PR fix?**

Fixes https://github.com/bazelbuild/rules_go/issues/2437

### CODE REVIEW

File: docs/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    out = "header.vm",
+    content = [
+        _NAV_HEADER,
+        "${moduleDocstring}",
+        _NAV_FOOTER,
+    ],
+)
+
+_DOC_SRCS = {
+    "extras:embed_data": "//go:extras.rst",
```
[the remark this answers is not in the document]
Comment (Cvarier) [reply docs/BUILD.bazel:37]: Yes, you're right, updating this.

**Comment by Cvarier** [comment]
@achew22 Thank you for your review! I addressed your comments and updated the PR again, let me know what you think!

File: extras/embed_data.bzl [hunk 2]
```diff
@@ -12,13 +12,56 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""embed_data.bzl provides the go_embed_data rule for embedding data in go files"""
+"""
+  [gazelle rule]: https://github.com/bazelbuild/bazel-gazelle#bazel-rule
```
Comment (achew22) [inline extras/embed_data.bzl:16]: Thanks for moving this in. I think we should still have the header start out with `# Extra rules`, but include a section after `## Contents` called `## Additional resources` that contains these 4 links

Comment (Cvarier) [reply extras/embed_data.bzl:16]: Sure thing, just updated

File: go/extras.rst [hunk 3]
```diff
@@ -1,95 +0,0 @@
-Extra rules
```
Comment (achew22) [inline go/extras.rst:1]: Please don't delete the file, instead leave a stub file that says "This file has moved, please look at it here"

**Review by achew22 — CHANGES_REQUESTED** [review]
Two little ones then I think we can merge this

**Comment by Cvarier** [comment]
@achew22 updated again

**Comment by achew22** [comment]
@Cvarier, thanks for the great PR! This is a big win for the project and progress for the whole ecosystem.

**Pull request merged**